core/src/wallet/ripple/transaction_builders/RippleLikeTransactionBuilder.cpp

```diff
@@ -158,6 +162,14 @@ namespace ledger {
             auto bigIntSequence = BigInt::fromHex(hex::toString(sequence));
             tx->setSequence(bigIntSequence);
 
+            auto destinationTagCode = reader.peek();
```

Comment:
I think a better way to do this would be to `reader.read` the bytes and `switch` or have branching with `if` / `else if`. The problem here is that if `destinationTagCode ≠ 0x2E`, you’re going to throw the peek away for nothing at line 174.

Reply:
Agreed on that, I complied with has been done before but I can change that !

Reply:
Actually I just checked the next field and it has 2 bytes, so we might lose in readability, here is the result of it : 
```
        auto nextCode = reader.readNextByte();
            switch (nextCode) {
                case 0x2E:
                    //1 byte Destination tag:   Type Code = 2, Field Code = 14
                    auto bigIntTag = BigInt::fromHex(hex::toString(reader.read(4)));
                    tx->setDestinationTag(static_cast<const uint32_t>(bigIntTag.toUint64()));
                    break;
                default:
                    break;
            }

            //2 bytes LastLedgerSequence Field ID:   Type Code = 2, Field Code = 27
            reader.readNextByte(); 
            //LastLedgerSequence
            auto ledgerSequence = reader.read(4);
```
So we lose the 2 bytes thing of LastLedgerSequence (refer to comment right above about field & type code)
@phaazon what do you think ?